src/shogun/lib/parameter_observers/ParameterObserverLogger.cpp

```diff
@@ -0,0 +1,33 @@
+/*
+* Written (W) 2019 Giovanni De Toni
+*/
+
+#include <shogun/lib/parameter_observers/ParameterObserverLogger.h>
+//#include <shogun/lib/type_case.h>
+
+using namespace shogun;
+
+ParameterObserverLogger::ParameterObserverLogger() {}
+
+ParameterObserverLogger::ParameterObserverLogger(std::vector<std::string> &parameters)
+		: ParameterObserverInterface(parameters) {}
+
+void ParameterObserverLogger::on_next(const TimedObservedValue &value) {
+
+	//auto f_print = [&value](auto _disp_value) {
+	//	SG_SPRINT("[%l] Received %s", convert_to_millis(value.second), _disp_value)
+	//};
+	// Dispatch the value and print it
+	//sg_any_dispatch(value.first.get_value(), sg_all_typemap, f_print, f_print, f_print);
```

Comment:
The original idea of this logger was to print on screen (or on file) the string representation of the observed value.
However, I do not know if it is possible within Shogun (do we have a sort of `to_string()` method for all Shogun's objects?) or if it is feasible without using hacky ways.

Reply:
we do! see for example #4537 where @gf712 changed it to have nice strings for enum/options

Reply:
For now, this printing is fine, but generally, we won't to have an abstraction layer for where the generated string is going (we can do that later)

Reply:
Cool! For now, I will leave it just a simple debug observer which will be also useful to check that we are getting the correct values out of the machines.